File: Python/7py 4/reverse_letter.py
"""
|--------------------------------------------------------------------------
| Simple Fun #176: Reverse Letter
|--------------------------------------------------------------------------
|
"""

import re
def reverse_letter(string):
   return ''.join(list(reversed(list(filter(lambda i: re.match("[a-z]", i) , string)))))


def reverse_letter(s):
   return ''.join([i for i in s if i.isalpha()])[::-1]


def reverse_letter(string):
   return ''.join(filter(str.isalpha, reversed(string)))


import re
def reverse_letter(string):
   return re.sub("[^a-zA-Z]","",string)[::-1]


def reverse_letter(s):
   return ''.join(filter(str.isalpha, s))[::-1]


print(reverse_letter("krishan"),"nahsirk")
print(reverse_letter("ultr53o?n"),"nortlu")
print(reverse_letter("ab23c"),"cba")
print(reverse_letter("krish21an"),"nahsirk")

File: JavaScript/7js 3/reverseLetter.js
/*
|--------------------------------------------------------------------------
| Simple Fun #176: Reverse Letter
|--------------------------------------------------------------------------
|
*/

function reverseLetter(str) {
   return str.split('').filter(i => i.match(/^[a-z]+$/)).reverse().join('')
}


const reverseLetter = (s) => s.replace(/[^a-z]/gi, '').split('').reverse().join('');
const reverseLetter = str => str.match(/[a-z]/g).reverse().join('');


function reverseLetter(str) {
   return str.match(/[a-z]/ig).reverse().join('');
}


function reverseLetter(str) {
   return str.split('').reverse().filter(function (el) {
      if ('abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ'.indexOf(el) != -1) {
         return el;
      }
   }).join('');
}


function reverseLetter(str) {
   return [...str].reduce((s, c) => /[A-Z]/i.test(c) ? c + s : s, "")
}


console.log(reverseLetter("krishan"), "nahsirk")
console.log(reverseLetter("ultr53o?n"), "nortlu")
console.log(reverseLetter("ab23c"), "cba")
console.log(reverseLetter("krish21an"), "nahsirk")